apps/src/templates/teacherDashboard/SectionRow.jsx

```diff
@@ -207,6 +215,12 @@ class SectionRow extends Component {
     } = this.props;
     const { editing, deleting } = this.state;
 
+    // When deleting a section, I've occasionally seen us attempt a render with
+    // a null section for some reason. This is here to provide defense against this.
+    if (!section) {
+      return null;
+    }
```

Comment:
I had considered that. I think I was trying to be in a place where changing a different row didn't result in a rerender of the current row. However, even as written, I'm not sure (a) that happens anyways without without implementing `componentShouldUpdate` and (b) its an optimization that really matters overmuch.

Assuming you're okay with it, I'd like to clean this up in a future PR (where I also have a few other smallish cleanups planned).